pallets/automation-time/src/lib.rs

```diff
@@ -955,6 +1089,80 @@ pub mod pallet {
 				.saturating_add(<T as Config>::WeightInfo::run_xcmp_task())
 		}
 
+		/// Executes auto compounding delegation and reschedules task on success
+		pub fn run_auto_compound_delegated_stake_task(
+			delegator: T::AccountId,
+			collator: T::AccountId,
+			account_minimum: BalanceOf<T>,
+			frequency: Seconds,
+			task_id: T::Hash,
+			mut task: Task<T>,
+		) -> (Task<T>, Weight) {
+			match Self::compound_delegator_stake(
+				delegator.clone(),
+				collator.clone(),
+				account_minimum,
+				Self::calculate_execution_fee(&task.action, 1),
+			) {
+				Ok(delegation) =>
+					Self::deposit_event(Event::SuccesfullyAutoCompoundedDelegatorStake {
+						task_id,
+						amount: delegation,
+					}),
+				Err(e) => {
+					Self::deposit_event(Event::AutoCompoundDelegatorStakeFailed {
+						task_id,
+						error_message: Into::<&str>::into(e).as_bytes().to_vec(),
+						error: e,
+					});
+					return (
+						task,
+						<T as Config>::WeightInfo::run_auto_compound_delegated_stake_task(),
+					)
+				},
+			}
+
+			let new_execution_times: Vec<UnixTime> =
+				task.execution_times.iter().map(|when| when.saturating_add(frequency)).collect();
+			let _ = Self::reschedule_existing_task(
```

Comment:
I'm not thrilled about my implementation here as it requires updating the task afterwards to ensure that all the metadata like `execution_times` and `executions_left` stays correct. In the long (hopefully short) run, I'm hoping that all of these methods that act on the task data can be moved to the task impl so that state management becomes easier. Anyway, I'm not thrilled with my implementation and very open to improvements.